docs/reference/commandline/build.md

```diff
@@ -489,6 +490,49 @@ FROM alpine AS production-env
 $ docker build -t mybuildimage --target build-env .
 ```
 
+### Custom build outputs
+
+By default, a local container image is created from the build result. With the `--output` flag this behavior can be overriden and a custom exporter can be used instead. For example, this is useful for generating local binaries, code generation etc.
+
+The value for `--output` is a CSV-formatted string defining the exporter type and options. Currently, `local` and `tar` exporters are supported. The local exporter writes the resulting build files to a directory on the client side. The tar exporter is similar but writes the files as a single tarball.
+
+If no type is specified, the value defaults to the output directory of the local exporter. `-` can be used to write the output tarball to standard output.
+
+Examples:
+
+```
+docker build -o out .
```

Comment:
These examples show what syntax is possible, but don't explain what each form does; perhaps better to split them up into separate blocks, and a short introduction, e.g.

```suggestion

The following example builds an image using the current directory (`.`) as build context,
and exports the files to a directory named `out` in the current directory. If the directory
does not exist, Docker creates the directory automatically:

```bash
docker build -o out .
```